File: a2sv/g51/codebase/leetcode/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int zeros = 0, n = nums.length, l = 0, ans = 0;

        for (int i = 0; i < n ; i++){
            if (nums[i] == 0) zeros++;
            while (l < n && zeros > 1){
                if (nums[l++] == 0) zeros--;
            }
            
            ans = Math.max(i -l, ans);
        }
        return ans;
        
    }
}

File: a2sv/g51/codebase/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums):
        zeros = 0
        n = len(nums)
        l = 0
        ans = 0

        for i in range(n):
            if nums[i] == 0:
                zeros += 1
            while l < n and zeros > 1:
                if nums[l] == 0:
                    zeros -= 1
                l += 1

            ans = max(i - l, ans)

        return ans
